packages/react-components/src/components/RichTextEditor/Plugins/CopyPastePlugin.tsx

```diff
@@ -75,23 +77,27 @@ export const handleBeforePasteEvent = (
  */
 export const handleInlineImage = (
   event: PluginEvent,
-  onInsertInlineImage?: (image: string, fileName: string) => void
+  onInsertInlineImage?: (imageAttributes: Record<string, string>, fileName?: string) => void
 ): void => {
   if (event.eventType === PluginEventType.BeforePaste && event.pasteType === 'normal' && onInsertInlineImage) {
     event.fragment.querySelectorAll('img').forEach((image) => {
       const clipboardImage = event.clipboardData.image;
-      const fileName = clipboardImage?.name || clipboardImage?.type.replace('/', '.') || 'image.png';
+      const fileName = clipboardImage?.name || clipboardImage?.type.replace('/', '.');
       // If the image src is an external url, call the onInsertInlineImage callback with the url.
       let imageUrl = image.src;
       if (image.src.startsWith('data:image/')) {
         const blobImage = _base64ToBlob(image.src);
         imageUrl = URL.createObjectURL(blobImage);
       }
 
-      onInsertInlineImage(imageUrl, fileName);
-
       image.src = imageUrl;
       image.alt = image.alt || 'image';
+      image.id = generateGUID();
```

Comment:
Could you clarify why we need id here?

Reply:
> Could you clarify why we need id here?

For Component Contosos, they will get all the imageAttributes in insert and remove callbacks and they can use the id as a reference to compare images.
We also use it internally such as in `getAddedInlineImages` to compare images in the content with previous images

Reply:
Can we add a comment that includes this information?